### PR TITLE
VFB-2 Add button hovering feature and used Mui Paper for Calendar

### PR DESCRIPTION
**Pull request by ChloeYiuSoftwire**
I have added button hovering feature and used Mui Paper for Calendar

- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've linted via `npm run lint`
    - can run `npm run lint_fix` to try and fix as any mistakes automatically as possible!
- [x] Make sure you've tested via `npm run test`

### CODE REVIEW

**Review by SamYap-Softwire — APPROVED** [review]
code looks fine :)

File: src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -55,7 +57,9 @@ const CalendarStyling = styled.div`
     .fc .fc-icon {
         font-size: min(4vw, 1.2rem);
     }
-
+    .fc .fc-scrollgrid-section-header.fc-scrollgrid-section-sticky > * {
```
Comment (paopaosoftwire) [inline src/components/Calendar/Calendar.tsx:60]: Can you explain what the > * is doing?

**Review by paopaosoftwire — APPROVED** [review]
Looking good!

File: src/components/Calendar/Calendar.tsx (hunk 1 above)
Comment (Var1377) [reply src/components/Calendar/Calendar.tsx:60]: that just means that it targets all direct descendants of `.fc-scrollgrid-section-header.fc-scrollgrid-section-sticky`

**Review by Var1377 — CHANGES_REQUESTED** [review]
image

Not having borders in the inactive cells looks odd in both dark mode and light mode, could you think of a better way to style it?

**Review by DanWilkins2107 — APPROVED** [review]
Am happy once other comments have been resolved!

**Comment by ChloeYiuSoftwire** [comment]
> image
> 
> Not having borders in the inactive cells looks odd in both dark mode and light mode, could you think of a better way to style it?

@Var1377 was a bug in firefox. Edited the code to make the border appear for those cells

**Review by Var1377 — APPROVED** [review]

File: src/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -28,7 +29,11 @@ export interface CalendarEvent {
     allDay?: boolean;
 }
 
-const CalendarStyling = styled.div`
+const CalendarStyling = styled(Paper)`
+    border-radius: 2rem;
+    padding: 1.5rem;
+    background-color: ${(props) => props.theme.main.background[0]};
```
[the remark this answers is not in the document]
Comment (ChloeYiuSoftwire) [reply src/components/Calendar/Calendar.tsx]: It was a mistake. Removed

File: src/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -121,6 +124,10 @@ const CalendarStyling = styled.div`
         background-color: ${(props) => props.theme.primary.background[1]};
         border-color: ${(props) => props.theme.primary.background[1]};
         color: ${(props) => props.theme.primary.foreground[1]};
+
+        &:hover {
+            opacity: 80%;
```
Comment (TomCusack-Softwire) [inline src/components/Calendar/Calendar.tsx:129]: I thought the themes change meant we shouldn't change opacity where possible, and just use a lighter colour?

File: src/components/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -136,10 +143,18 @@ const CalendarStyling = styled.div`
     }
 
     .fc .fc-day-other {
-        background-color: ${(props) => props.theme.main.background[1]};
+        > div {
+            background-color: ${(props) => props.theme.main.background[2]};
```
Comment (TomCusack-Softwire) [inline src/components/Calendar/Calendar.tsx:147]: should this be using background2 without the color being foreground2?
and should the color be moved to within this too?

Comment (ChloeYiuSoftwire) [reply src/components/Calendar/Calendar.tsx:147]: Make sense. Refactored

File: src/components/Calendar/Calendar.tsx (hunk 3 above)
Comment (ChloeYiuSoftwire) [reply src/components/Calendar/Calendar.tsx:129]: Sure. Changed to using `primary[2]`

**Review by TomCusack-Softwire — APPROVED** [review]

**Pull request merged**